projects/ADuCM3029_demo_cn0415/src/platform_drivers.c

```diff
@@ -0,0 +1,741 @@
+
+	/* Enable output driver */
+	ret = adi_gpio_OutputEnable(port, pin, true);
+	if(ret != ADI_GPIO_SUCCESS)
+		return ret;
+
+	/* Initialize pin with a value */
+	if(value == 1)
+		return adi_gpio_SetHigh(port, pin);
+	else
+		return adi_gpio_SetLow(port, pin);
+}
+
+/**
+ * @brief Set the value of the specified GPIO.
+ * @param desc - The GPIO descriptor.
+ * @param value - The value.
+ *                Example: GPIO_HIGH
+ *                         GPIO_LOW
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t gpio_set_value(gpio_desc *desc, uint8_t value)
+{
+	uint16_t pin;
+	uint8_t port;
+
+	/* Get pin port and number from pin number */
+	pin = (ADI_GPIO_DATA)(0x0001 << (desc->number & 0x0F));
+	port = (desc->number & 0xF0) >> 4;
+
+	/* Change pin value */
+	if(value == 1)
+		return adi_gpio_SetHigh(port, pin);
+	else
+		return adi_gpio_SetLow(port, pin);
+}
+
+/**
+ * @brief Get the value of the specified GPIO.
+ * @param desc - The GPIO descriptor.
+ * @param value - The value.
+ *                Example: GPIO_HIGH
+ *                         GPIO_LOW
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t gpio_get_value(gpio_desc *desc, uint16_t *value)
+{
+	uint16_t pin;
+	uint8_t port;
+
+	/* Get pin port and number from pin number */
+	pin = (ADI_GPIO_DATA)(0x0001 << (desc->number & 0x0F));
+	port = (desc->number & 0xF0) >> 4;
+
+	/* Read pin value */
+	return adi_gpio_GetData(port, pin, value);
+}
+
+/**
+ * @brief Generate miliseconds delay.
+ * @param msecs - Delay in miliseconds.
+ * @return None.
+ */
+void mdelay(uint32_t msecs)
+{
+	timer_sleep(msecs);
+}
+
+/***************************************************************************//**
+ * @brief Initialize and start UART communication and UART controller descriptor
+ * 		  structure.
+ *
+ * @param descriptor - User UART device structure descriptor.
+ * @param init_param - Initilaization parameters.
+ *
+ * @return CN415_SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t usr_uart_init(struct uart_desc **descriptor,
+		      struct uart_init_param init_param)
+{
+	/* Variable for storing the return code from UART device */
+	ADI_UART_RESULT  e_uart_result;
+	uint8_t bits_nr;
+	struct uart_desc *desc;
+
+	desc = (struct uart_desc *)malloc(sizeof(*desc));
+	if (!desc)
+		return FAILURE;
+
+	desc->baudrate = init_param.baudrate;
+	desc->bits_no = init_param.bits_no;
+
+	/* Values for the baudrate generation registers when the baudrate is one of
+	 * the standard values and clock is 26MHz */
+	const uint8_t baudrate_osr_26mhz[] = {3, 3, 3, 3, 3, 3, 3, 2, 2, 2};
+	const uint16_t baudrate_divc_26mhz[] = {24, 12, 8, 4, 4, 2, 1, 1, 1, 1};
+	const uint8_t baudrate_divm_26mhz[] = {3, 3, 2, 3, 1, 1, 1, 1, 1, 1};
+	const uint16_t baudrate_divn_26mhz[] = {1078, 1078, 1321, 1078, 1563, 1563,
+						1563, 1563, 1280, 171
+					       };
```

Comment:
nitpick: this could be placed on the previous line.

Reply:
The AStyle places it this way.

Reply:
AStyle is not the king of the world :). We should take a second look at it's fixes.